### PR TITLE
fix storage account name prefix issue when using Azure CAPI install

### PR DESCRIPTION
**Pull request by jinyunma**
https://issues.redhat.com/browse/OCPQE-24713

Starting from 4.17, the default install method is CAPI-based, prefix of storage account name is changed to use infraId instead of "cluster".

### CODE REVIEW

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-multi-nightly-azure-ipi-amd-mixarch-f28-destructive

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-azure-ipi-arm-mixarch-f5

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-multi-nightly-azure-ipi-amd-mixarch-f28-destructive

File: ci-operator/step-registry/ipi/install/heterogeneous/ipi-install-heterogeneous-commands.sh [hunk 1]
```diff
@@ -239,6 +239,7 @@ case $CLUSTER_TYPE in
   infra_id=$(yq-v4 '.infraID' < "${SHARED_DIR}"/metadata.json)
   rg_name="${infra_id}-rg"
   sa_name=$(az storage account list -g "${rg_name}" | yq-v4 '.[] | select(.name == "cluster*").name')
+  [[ -z "${sa_name}" ]] && sa_name=$(az storage account list -g "${rg_name}" | yq-v4 ".[] | select(.name == \"${infra_id//-}*\").name")
```
Comment (aleskandro) [inline ci-operator/step-registry/ipi/install/heterogeneous/ipi-install-heterogeneous-commands.sh:242]: hi, @jinyunma, can you add some comments here to explain why we need this now for future readers?

Comment (jinyunma) [reply ci-operator/step-registry/ipi/install/heterogeneous/ipi-install-heterogeneous-commands.sh:245]: sure, updated.

**Comment by jinyunma** [comment]
Tested [4.16-multi-nightly-azure-ipi-arm-mixarch-f5
](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/55453/rehearse-55453-periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-azure-ipi-arm-mixarch-f5/1822950231361196032) and [4.17-multi-nightly-azure-ipi-amd-mixarch-f28-destructive with featureGates enabled](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/55453/rehearse-55453-periodic-ci-openshift-openshift-tests-private-release-4.17-multi-nightly-azure-ipi-amd-mixarch-f28-destructive/1822990078570401792), the step `ipi-install--heterogeneous` in both jobs were passed.

/pj-rehearse ack

cc @aleskandro @lwan-wanglin to review

**Comment by lwan-wanglin** [comment]
thanks, looks good to me
/lgtm

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-multi-nightly-azure-ipi-amd-mixarch-f28-destructive

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-multi-nightly-azure-ipi-amd-mixarch-f28-destructive

**Comment by jinyunma** [comment]
Based on the fix in [OCPBUGS-37953](https://issues.redhat.com/browse/OCPBUGS-37953), I updated the code accordingly, the step `ipi-install--heterogeneous` is successful in [testing job](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/55453/rehearse-55453-periodic-ci-openshift-openshift-tests-private-release-4.17-multi-nightly-azure-ipi-amd-mixarch-f28-destructive/1823542688922734592).

@aleskandro @lwan-wanglin please help to review again, thanks!

**Review by aleskandro — APPROVED** [review]
Hi @jinyunma thanks for this PR

/lgtm
/approve

**Pull request merged**